Tests/ApolloCodegenTests/CodeGeneration/Templates/OperationDefinitionTemplateTests.swift

```diff
@@ -0,0 +1,227 @@
+import XCTest
+import Nimble
+import OrderedCollections
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class OperationDefinitionTemplateTests: XCTestCase {
+
+  var schemaSDL: String!
+  var document: String!
+  var ir: IR!
+  var operation: IR.Operation!
+  var config: ApolloCodegenConfiguration!
+  var subject: OperationDefinitionTemplate!
+
+  override func setUp() {
+    super.setUp()
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    type Animal {
+      species: String!
+    }
+    """
+
+    document = """
+    query TestOperation {
+      allAnimals {
+        species
+      }
+    }
+    """
+
+    config = .mock()
+  }
+
+  override func tearDown() {
+    schemaSDL = nil
+    document = nil
+    ir = nil
+    operation = nil
+    config = nil
+    subject = nil
+    super.tearDown()
+  }
+
+  // MARK: - Helpers
+
+  func buildSubjectAndOperation(named operationName: String = "TestOperation") throws {
+    ir = try .mock(schema: schemaSDL, document: document)
+    let operationDefinition = try XCTUnwrap(ir.compilationResult[operation: operationName])
+    operation = ir.build(operation: operationDefinition)
+    subject = OperationDefinitionTemplate(
+      operation: operation,
+      schema: ir.schema,
+      config: config
+    )
+  }
+
+  // MARK: - Import Statements
+
+  func test__generate__givenFileOutput_inSchemaModule_schemaModuleManuallyLinked_generatesImportNotIncludingSchemaModule() throws {
+    // given
+    config = .mock(output: .mock(
+      moduleType: .manuallyLinked(namespace: "TestModuleName"),
+      operations: .inSchemaModule
+    ))
+
+    let expected =
+    """
+    import ApolloAPI
+
+    """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let actual = subject.render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, ignoringExtraLines: true))
+  }
+
+  func test__generate__givenFileOutput_inSchemaModule_schemaModuleNotManuallyLinked_generatesImportNotIncludingSchemaModule() throws {
+    // given
+    config = .mock(output: .mock(
+      moduleType: .swiftPackageManager(moduleName: "TestModuleName"),
+      operations: .inSchemaModule
+    ))
+
+    let expected =
+    """
+    import ApolloAPI
+
+    """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let actual = subject.render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, ignoringExtraLines: true))
+  }
+
+  func test__generate__givenFileOutput_notInSchemaModule_schemaModuleNotManuallyLinked_generatesImportIncludingSchemaModule() throws {
```

Comment:
We'll have to remember to expand these tests to include the other dependency managers once they're built.